module-sys/SystemManager/CpuSentinel.cpp

```diff
@@ -38,9 +38,29 @@ namespace sys
 
     void CpuSentinel::CpuFrequencyHasChanged(bsp::CpuFrequencyHz newFrequency)
     {
+        currentFrequency = newFrequency;
         if (callback) {
             callback(newFrequency);
         }
+        if (taskHandle != nullptr && newFrequency >= currentFrequencyToHold) {
+            xTaskNotifyGive(taskHandle);
+            taskHandle = nullptr;
+        }
     }
 
+    bool CpuSentinel::HoldMinimumFrequencyAndWait(bsp::CpuFrequencyHz frequencyToHold,
+                                                  TaskHandle_t taskToNotify,
+                                                  uint32_t timeout)
+    {
+        auto msg = std::make_shared<sys::HoldCpuFrequencyMessage>(GetName(), frequencyToHold);
+        owner->bus.sendUnicast(std::move(msg), service::name::system_manager);
+
+        if (currentFrequencyToHold < frequencyToHold) {
```

Comment:
it seems to me that there should be a condition: `if (currentFrequency < frequencyToHold) {` instead

Reply:
done